### ISSUE TITLE
[prefer-regexp-test] should not apply when the regex has the global modifier (g)

### ISSUE THREAD

**Issue by nikonthethird** [sindresorhus/eslint-plugin-unicorn#1114]
The rule [prefer-regexp-test] suggests that it is equivalent to replace a `match` method call on a string to a `test` method call on a regex when the `match` call is only ever used to see if the regex matches.

This is only correct when the regex in question does not have the global modifier (g). I added an example at the bottom where you can see a different result for the second call to `test`. [The example is also available as a fiddle here.](https://jsfiddle.net/0dbkq5v6/)

You can also see another example in the [MDN documentation on `test`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test).

Removing the global modifier in the example below fixes the issue.

```js
const regex = /weird/g;

const testString = 'well this is weird!';

console.log(`First match execution: ${!!testString.match(regex)}`);
// Prints: First match execution: true

console.log(`Second match execution: ${!!testString.match(regex)}`);
// Prints: Second match execution: true

console.log(`First test execution: ${regex.test(testString)}`);
// Prints: First test execution: true

console.log(`Second test execution: ${regex.test(testString)}`);
// Prints: Second test execution: false
```
